### PR TITLE
docs: init a vuepress documentation

### PR DESCRIPTION
**Pull request by antleblanc**
## :memo: Documentation

This Pull Request bring a documentation to easily started with the Monorepo.

It is powered by [VuePress](https://vuepress.vuejs.org/).

910fc87 - docs: init a vuepress documentation

## :construction_worker_man: Build

7d421bb - build: add vuepress:deploy command

## :bug: Bug Fix

ef0145d - fix: add missing repository field to the package.json template
49c5384 - fix: add missing repository field to the package.json template

## :eyes: Preview

A preview is availabe here: https://antleblanc.github.io/playground (not fully up-to-date).

## :camera_flash: Screenshots

home

getting-started

## :bookmark: TODOs

- [x] Add deploy command.

## :house: Internal

- No quality check required.

### CODE REVIEW

File: docs/guide/applications.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+| Application                   | Region       |
+| ----------------------------- | ------------ |
+| [Web](#web)                   | EU           |
+| [Server](#server)             | EU / CA / US |
+| [Public Cloud](#public-cloud) | EU / CA / US |
+| [Telecom](#telecom)           | EU           |
+
+:::tip Information
+Pay attention that regarding the region, you should rely on a different API URL.
```
Comment (frenautvh) [inline docs/guide/applications.md]: -regarding +depending

File: docs/guide/architecture.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Architecture
+
+## General information
+
+The monorepo is powered by two technologies:
```
Comment (frenautvh) [inline docs/guide/architecture.md]: i would just say "The monorepo is powered by :", not sure if technologies is the right word :)

**Comment by frenautvh** [comment]
looks good :)

**Comment by FredericEspiau** [comment]
What files are yours and what files are not ? We are not going to review generated files 😄

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      href="https://ovh.com/manager/"
+      rel="noopener noreferrer"
+      target="_blank">
+      ovh.com/manager
+      <icon-external-link/>
+    </a>
+  </div>
+  <div class="feature">
+    <h2>Website</h2>
+    <p>Getting more informations about our products, please visit our website.</p>
```
Comment (FredericEspiau) [inline docs/README.md]:
```suggestion
    <p>To have more informations about our products, please visit our website.</p>
```

File: docs/guide/applications.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+# Applications
+
+All applications are located in `packages/manager/apps/*` workspace.
```
Comment (FredericEspiau) [inline docs/guide/applications.md]:
```suggestion
All applications are located in the `packages/manager/apps/*` workspace.
```

File: docs/guide/applications.md (hunk 1 above)
Comment (FredericEspiau) [reply docs/guide/applications.md]: `depending on`

Comment (FredericEspiau) [reply docs/guide/applications.md]:
```suggestion
Depending on the requested region, the URL to access the API will be different
```

File: docs/guide/applications.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+### How to start the application?
+
+#### Using `yarn workspace` <Badge text="recommended"/>
+
+```sh
+$ yarn workspace @ovh-ux/manager-web run start:dev
+```
+
+#### Using alternative:
```
Comment (FredericEspiau) [inline docs/guide/applications.md]:
```suggestion
#### Using alternative
```
?

File: docs/guide/architecture.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Languages and Framework
+
+The **Manager** application is using these following languages:
+
+- HTML
+- CSS ([Less.js](http://lesscss.org/), [SCSS](https://sass-lang.com/))
+- JavaScript
+
+[AngularJS](https://angularjs.org/) is widely used as the MVW Framework to build [Single Page Applications](https://en.wikipedia.org/wiki/Single-page_application).
```
Comment (FredericEspiau) [inline docs/guide/architecture.md]:
```suggestion
[AngularJS](https://angularjs.org/) used to be a widelys used MVW Framework to build [Single Page Applications](https://en.wikipedia.org/wiki/Single-page_application).
```

File: docs/guide/architecture.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+::: tip Information
+We are exploring some new JavaScript Frameworks to speed up the development workflow.
```
Comment (FredericEspiau) [inline docs/guide/architecture.md]:
```suggestion
We are exploring some new solutions to speed up the development workflow.
```

File: docs/guide/tools.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Tools
+
+::: warning TODOs
+:warning: This section is in a work in progress :warning:
```
Comment (FredericEspiau) [inline docs/guide/tools.md]:
```suggestion
:warning: This section is a work in progress :warning:
```

File: packages/manager/apps/overthebox/package.json [hunk 9]
```diff
@@ -2,6 +2,11 @@
   "name": "@ovh-ux/manager-overthebox-app",
   "version": "5.0.0",
   "private": true,
+  "repository": {
```
Comment (FredericEspiau) [inline packages/manager/apps/overthebox/package.json]: What does this have to do with this PR ?

File: docs/how-to/README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
+---
+sidebar: auto
+---
+
+# How To
+
+## Install Git
+
+You can download the Git installer corresponding to your system [here](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
```
Comment (FredericEspiau) [inline docs/how-to/README.md]: Some users might not want to use installers but package managers to install Git, it would be easier to say

```suggestion
You can find the documention to install Git there [here](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
```

File: docs/.vuepress/components/ListPackages.vue [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        v-for="pkg in packages"
+        v-bind:key="pkg.package.name">
+        <a
+          v-bind:href="'https://github.com/ovh/manager/tree/master/' + pkg.package.repository.directory"
+          rel="noopener noreferrer"
+          target="_blank">
+          {{ pkg.package.name }}@{{ pkg.package.version }}
+        </a>
+        <br>
+        <small>
```
Comment (FredericEspiau) [inline docs/.vuepress/components/ListPackages.vue]: `small` has a semantic meaning, it is used for small prints, this is not the case here

File: docs/guide/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
+# Getting started
+
+## Prerequisites
+
+- [Git](https://git-scm.com)
+- [Node.js](https://nodejs.org/en/) >= 12.13
```
Comment (FredericEspiau) [inline docs/guide/getting-started.md]: If this is hardcoded, this won't be updated 😄

File: docs/guide/applications.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
+# Applications
```
Comment (FredericEspiau) [inline docs/guide/applications.md]: When I arrive on this screen, I get scrolled to the bottom of the screen

File: docs/guide/modules.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
+# Modules
+
+::: warning TODOs
+:warning: This section is in a work in progress :warning:
```
Comment (FredericEspiau) [inline docs/guide/modules.md]:
```suggestion
:warning: This section is a work in progress :warning:
```

Comment (FredericEspiau) [reply docs/guide/modules.md]: (can't you componentize this ?)

File: docs/guide/modules.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
+# Modules
```
Comment (FredericEspiau) [inline docs/guide/modules.md]: Same issue, I get scrolled to the bottom of the screen

**Comment by FredericEspiau** [comment]
Good work 👍

File: packages/manager/apps/overthebox/package.json (hunk 9 above)
Comment (antleblanc) [reply packages/manager/apps/overthebox/package.json]: The `repository` entry is needed to be able to display all components
and associated them to their corresponding location.

File: docs/.vuepress/components/ListPackages.vue (hunk 11 above)
Comment (antleblanc) [reply docs/.vuepress/components/ListPackages.vue]: It is! I don't want to display a the package description as the same font Size than its name.

File: docs/guide/getting-started.md (hunk 12 above)
Comment (antleblanc) [reply docs/guide/getting-started.md]: Same with the root `README.md` file.

When **Prerequisites** will evolve then we need to update it here too.

File: docs/guide/applications.md (hunk 13 above)
Comment (antleblanc) [reply docs/guide/applications.md]: Caution: the preview link isn't up-to-date.

I will take a closer look if I'm able to reproduce.

File: docs/guide/modules.md (hunk 14 above)
Comment (antleblanc) [reply docs/guide/modules.md]: Sure! But don't want to let them in place for a long period.

Best option is to finish these sections.

File: docs/.vuepress/components/Contributing.vue [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    };
+  },
+  async mounted () {
+    try {
+      const contributing = await fetch('https://raw.githubusercontent.com/ovh/manager/master/CONTRIBUTING.md');
+      const text = await contributing.text();
+
+      this.content = md().render(text);
+    } catch (error) {
+      console.log('error', error);
```
Comment (marie-j) [inline docs/.vuepress/components/Contributing.vue]: Do we display a placeholder or something if we can't fetch the file ?

File: docs/.vuepress/components/ListPackages.vue [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    </div>
+
+    <p v-if="rejected">
+      Unable to get the packages list.
+    </p>
+
+    <ul v-if="success">
+      <li
+        v-for="pkg in packages"
+        v-bind:key="pkg.package.name">
```
Comment (marie-j) [inline docs/.vuepress/components/ListPackages.vue]: ❤️

File: docs/.vuepress/components/ListPackages.vue [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+      break;
+    case 'modules':
+      return 'packages/manager/modules/*';
+      break;
+    case 'tools':
+      return 'packages/manager/tools/*';
+      break;
+    default:
+      return 'packages/manager/apps/*';
+      break;
```
Comment (marie-j) [inline docs/.vuepress/components/ListPackages.vue]:
```js
switch (type) {
  case 'modules':
  case: 'tools':
  case: 'apps': 
    return `packages/manager/${type}/*`
  default:
   return `packages/components/*`
```
?

File: docs/.vuepress/components/ListPackages.vue [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+      rejected: false,
+      packages: [],
+    };
+  },
+  async mounted () {
+    this.loading = true;
+    try {
+      const packages = await fetch('/manager/assets/json/packages.json');
+      const pacakgesAsJson = await packages.json();
+      const { packagesList } = pacakgesAsJson.find((pkg) => pkg.workspace === getWorkspace(this.type));
```
Comment (marie-j) [inline docs/.vuepress/components/ListPackages.vue]:
```suggestion
      const { packagesList } = packagesAsJson.find((pkg) => pkg.workspace === getWorkspace(this.type));
```

**Comment by antleblanc** [comment]
> What files are yours and what files are not ? We are not going to review generated files

The only generated files in this Pull Request are:
- docs/.vuepress/public/assets/json/packages.json
- yarn.lock

To init this documentation I've added the `repository` to all `package.json` file.

I plan to review all of them to add some meta information. It will be helpful for
the `<ListPackages>` component. In that way, we will be able to display them
with a lot a useful information.

File: docs/.vuepress/components/ListPackages.vue (hunk 11 above)
Comment (FredericEspiau) [reply docs/.vuepress/components/ListPackages.vue]: `fine prints` means `legal text`, not `small characters` 😅

File: docs/guide/applications.md [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+
+All applications are located in the `packages/manager/apps/*` workspace.
+
+It houses:
+
+- Four large AngularJS monolithic applications (listed below).
+- A Sign-up form.
+- Several standalone applications.
+
+An application can be started for a specific region.
```
Comment (marie-j) [inline docs/guide/applications.md]:
```suggestion
The monolithic applications can be started for a specific region.
```

File: docs/guide/modules.md (hunk 14 above)
Comment (FredericEspiau) [reply docs/guide/modules.md]: We could need it in the future though

File: docs/guide/getting-started.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+- [Yarn](https://yarnpkg.com/lang/en/) >= 1.21.1
+- Supported OSes: GNU/Linux, macOS and Windows
```
Comment (JeremyDec) [inline docs/guide/getting-started.md]:
```suggestion
- Supported OS: GNU/Linux, macOS and Windows
```

File: packages/manager/modules/web-universe-components/package.json [hunk 22]
```diff
@@ -2,6 +2,11 @@
   "name": "@ovh-ux/ng-ovh-web-universe-components",
   "version": "7.2.1",
   "private": true,
+  "repository": {
```
Comment (marie-j) [inline packages/manager/modules/web-universe-components/package.json]: Maybe this should be added by default when crafting a module with the cli

File: docs/guide/applications.md (hunk 13 above)
Comment (JeremyDec) [reply docs/guide/applications.md]: Like @FredericEspiau, I have same behavior on each page, once you've scrolled down one

File: docs/guide/applications.md [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
+```sh
+$ yarn run build -p @ovh-ux/manager-dedicated # or @ovh-ux/manager-cloud
+```
+
+## Public Cloud
+
+![](/manager/assets/img/control-panel-public-cloud.jpg)
+
+::: tip Information
+Application can be started in different region with a given environment variable.
```
Comment (JeremyDec) [inline docs/guide/applications.md]:
```suggestion
Application can be started in different regions with a given environment variable.
```
Or `Application can be started in a different region with a given environment variable.`

File: docs/.vuepress/components/Contributing.vue (hunk 16 above)
Comment (antleblanc) [reply docs/.vuepress/components/Contributing.vue]: It requires me to check the status code. I will take a closer look.

File: docs/guide/applications.md (hunk 13 above)
Comment (antleblanc) [reply docs/guide/applications.md]: @FredericEspiau @JeremyDec Found the issue.

I will update the preview link.

File: docs/README.md [hunk 24]
```diff
@@ -0,0 +1,46 @@
+---
+home: true
+actionText: Get Started →
```
Comment (FredericEspiau) [inline docs/README.md]:
```suggestion
actionText: Get started →
```
no ?

File: docs/.vuepress/config.js [hunk 25]
```diff
@@ -0,0 +1,79 @@
+module.exports = {
+  base: '/manager/',
+  title: 'Manager',
+  description: 'OVHcloud Control Panel Documentation',
```
[the remark this answers is not in the document]
Comment (antleblanc) [reply docs/.vuepress/config.js]: This description is displayed to the homepage and it's already a bit long in my opinion.

Other suggestion:

```txt
Documentation for the Control Panel
```

Is it ok for you?

File: docs/README.md (hunk 24 above)
Comment (antleblanc) [reply docs/README.md]: I prefer the previous one.

File: docs/.vuepress/config.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (FredericEspiau) [reply docs/.vuepress/config.js]: Yup why not 🙂
